clang/include/clang/ASTMatchers/ASTMatchers.h

```diff
@@ -5911,6 +5911,30 @@ AST_POLYMORPHIC_MATCHER_P(hasAnySubstatement,
                                           Builder) != CS->body_end();
 }
 
+/// Matches compound statements that contain adjacent substatements matching
```

Comment:
Matches compound statements where at least one set of adjacent substatements matching the provided sequence of matchers.